src/Handler/StreamHandler.php

```diff
@@ -277,6 +278,7 @@ private function createStream(RequestInterface $request, array $options)
 
         $params = [];
         $context = $this->getDefaultContext($request, $options);
+        $uri = $request->getUri()->withFragment('');
```

Comment:
In fact it's not a good idea IMO as the reference for this object will not be the same as you expected.